### ISSUE TITLE
busted.runner fails when running standalone

### ISSUE THREAD

**Issue by nFec** [Olivine-Labs/busted#502]
Fresh busted installation fails with
```
lua: ...files (x86)\lua\5.1\\share\lua\5.1\busted\runner.lua:162: attempt to index local 'file' (a nil value)
```

The relevant section in runner.lua is
```Lua
-- If running standalone, setup test file to be compatible with live coding
  if options.standalone then
    local ctx = busted.context.get()
    local children = busted.context.children(ctx)
    local file = children[#children]
    debug.getmetatable(file.run).__call = info.func
  end
```

Minimal example:

busted_test.lua
```Lua
require 'busted.runner' ()
```
Just this one line

execute with
```
D:\lua busted_test.lua
```
Environment:
Win7-64bit
```
D:\>lua -v
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio

D:\>lua busted_test.lua --version
2.0.rc11-0
```

**Comment by daurnimator** [comment]
Same trying to run from command line on linux:
```
$ lua 
Lua 5.3.2  Copyright (C) 1994-2015 Lua.org, PUC-Rio
> require"busted.runner"()
/usr/share/lua/5.3/busted/runner.lua:162: attempt to index a nil value (local 'file')
stack traceback:
	/usr/share/lua/5.3/busted/runner.lua:162: in function 'busted.runner'
	(...tail calls...)
	[C]: in ?
```

**Comment by o-lim** [comment]
You are running it wrong on linux.  You cannot do `require 'busted.runner'()` directly from the interactive lua interpreter, it must be inside a lua file.  This works on linux, and there is a unit test for it which executes `spec/cl_standalone.lua`, which runs in travis and passes.  I am also able to run this just fine on Win7-64bit.  Maybe there is something wrong with your lua installation on Windows?

**Comment by daurnimator** [comment]
> You cannot do require 'busted.runner'() directly from the interactive lua interpreter, it must be inside a lua file.

why not? lets make it work :)

**Comment by o-lim** [comment]
I don't think this is possible since there is no way to detect if you are running Lua in the interactive interpreter or as a non-interactive script.  Furthermore, the call to `require 'busted.runner'()` does not return, it will exit once it has run all of your tests, and since you have not written any tests yet, it will exit immediately reporting the status for running 0 tests.

**Comment by DorianGray** [comment]
Maybe we can use the debug library to look up the stack and see if there's a "require" call, then grab the contents of that file using the same and loadstring it in a busted context? We'd need to probably do this after "exec"ing into a real busted runner to avoid the rest of the test being executed.

Messy, but might work? I'm not sure about maintainability but it seems it could be done in a semi-elegant way.

**Comment by o-lim** [comment]
That's basically what it is already doing.  It grabs the function that called `require 'busted.runner'` using the debug library and creates a file context using that function (this actually works really well with zbsutdio's live coding feature).  However, when you are in the interactive Lua interpreter, that function is basically empty since the tests are yet to be written.  When running in non-interactive mode, this is not a problem since everything is already in a file or being redirected from a file to stdin.  The problem is running in interactive mode.  In such a case, the easiest option I see is to pass a function like so
```lua
require 'busted.runner'(function()
  describe('Tests in interactive mode', function()
    it('check some stuff', function()
       ...
    end)
    it('check some more stuff', function()
       ...
    end)
  end)
 end)
```
or
```lua
run = require 'busted.runner'
... -- do some more "requires"
run(function()
  describe('Tests in interactive mode', function()
    it('check some stuff', function()
       ...
    end)
    it('check some more stuff', function()
       ...
    end)
  end)
end)
```
Then you would have to consider that this could be done multiple times in the interpreter, not just once.  But, if you're going to go through all the work of typing all of that into the interpreter, you might as well type it into a file instead.
